### PR TITLE
feat: retry failed update-lock builds

### PR DESCRIPTION
**Pull request by SurferJeffAtGoogle**
One step towards fixing https://github.com/googleapis/repo-automation-bots/issues/1777.

### CODE REVIEW

File: packages/owl-bot/src/scan-and-retry-failed-lock-updates.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+export async function scanAndRetryFailedLockUpdates(
+  projectId: string,
+  triggerId: string,
+  maxTries: number
+): Promise<void> {
+  const cb = core.getCloudBuildInstance();
+  const builds = cb.listBuildsAsync({projectId, pageSize: 100});
+  const rebuilds = await filterBuildsToRetry(triggerId, maxTries, builds);
+  console.info(`${rebuilds.length} to rebuild`);
```
Comment (sofisl) [inline packages/owl-bot/src/scan-and-retry-failed-lock-updates.ts:31]: Should this be logger.info?

Comment (sofisl) [reply packages/owl-bot/src/scan-and-retry-failed-lock-updates.ts:31]: same for below

File: packages/owl-bot/src/scan-and-retry-failed-lock-updates.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+/**
+ * Scans all the cloud builds for the past 24 hours looking for failures for
+ * the given build trigger.  Returns a list of builds that should be retried
+ * because they never succeeded.
+ */
+export async function filterBuildsToRetry(
+  triggerId: string,
+  maxTries: number,
```
Comment (sofisl) [inline packages/owl-bot/src/scan-and-retry-failed-lock-updates.ts:50]: what is maxTries? Should it be a constant?

File: packages/owl-bot/src/scan-and-retry-failed-lock-updates.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  projectId: string,
+  triggerId: string,
+  maxTries: number
+): Promise<void> {
+  const cb = core.getCloudBuildInstance();
+  const builds = cb.listBuildsAsync({projectId, pageSize: 100});
+  const rebuilds = await filterBuildsToRetry(triggerId, maxTries, builds);
+  console.info(`${rebuilds.length} to rebuild`);
+  for (const build of rebuilds) {
+    console.info(`Triggering rebuild for ${build.name}`);
```
Comment (sofisl) [inline packages/owl-bot/src/scan-and-retry-failed-lock-updates.ts:33]: Should there be any timeouts before rebuilding?

File: packages/owl-bot/src/scan-and-retry-failed-lock-updates.ts (hunk 1 above)
Comment (SurferJeffAtGoogle) [reply packages/owl-bot/src/scan-and-retry-failed-lock-updates.ts]: Done.

File: packages/owl-bot/src/scan-and-retry-failed-lock-updates.ts (hunk 2 above)
Comment (SurferJeffAtGoogle) [reply packages/owl-bot/src/scan-and-retry-failed-lock-updates.ts:67]: Added comment.

File: packages/owl-bot/src/scan-and-retry-failed-lock-updates.ts (hunk 3 above)
Comment (SurferJeffAtGoogle) [reply packages/owl-bot/src/scan-and-retry-failed-lock-updates.ts]: Added comment.

**Review by sofisl — APPROVED** [review]

**Pull request merged**